pkg/eval/str/str.go

```diff
@@ -8,6 +8,316 @@ import (
 	"github.com/elves/elvish/pkg/eval"
 )
 
+//elvdoc:fn compare
+//
+// ```elvish
+// str:compare $a $b
+// ```
+//
+// Compare returns an integer comparing two strings lexicographically. The
+// result will be 0 if a==b, -1 if a < b, and +1 if a > b.
+//
+// ```elvish-transcript
+// ~> str:compare a a
+// > 0
+// ~> str:compare a b
+// > -1
+// ~> str:compare b a
+// > 1
+// ```
+
+//elvdoc:fn contains
+//
+// ```elvish
+// str:contains $str $substr
+// ```
+//
+// Contains returns `$true` if `$substr` is within `$str` else `$false`.
```

Comment:
This is kind of hard to parse. I know you are trying to be clear what this returns, but I like how Go's wording "reports whether" makes it clear that this is a predicate, so I'd say stick to the original wording. Same elsewhere.

Reply:
Done, but in an elvish fashion.